### PR TITLE
Add check in sync when report is not provided by Adviser

### PR DESCRIPTION
**Pull request by pacospace**
Signed-off-by: Francesco Murdaca <fmurdaca@redhat.com>

## This introduces a breaking change

- [ ] Yes
- [x] No

## This should yield a new module release

- [x] Yes
- [ ] No

## This Pull Request implements

Add check in adviser sync when report is not provided by Adviser document

### CODE REVIEW

File: thoth/storages/graph/postgres.py [hunk 1]
```diff
@@ -4548,46 +4548,51 @@ def sync_adviser_result(self, document: dict) -> None:
...
-                performance_score = None
-                overall_score = product["score"]
-                for entry in product.get("justification", []):
-                    if "performance_score" in entry:
-                        if performance_score is not None:
-                            _LOGGER.error(
-                                "Multiple performance score entries found in %r (index: %d)", adviser_document_id, idx
-                            )
-                        performance_score = entry["performance_score"]
+            if document["result"].get("report", {}):
```
Comment (fridex) [inline thoth/storages/graph/postgres.py]: To reduce code complexity:

```python
            if not document["result"].get("report"):
                return
```

Comment (pacospace) [reply thoth/storages/graph/postgres.py]: Thanks! and leave the warning?

Comment (fridex) [reply thoth/storages/graph/postgres.py]: Yes. :+1:

File: thoth/storages/graph/postgres.py [hunk 2]
```diff
@@ -4548,6 +4548,12 @@ def sync_adviser_result(self, document: dict) -> None:
             )
 
             # Output stacks - advised stacks
+            if not document["result"].get("report", {}):
```
Comment (fridex) [inline thoth/storages/graph/postgres.py]: Also, the sync of unsolved packages will need to go before this if statement - there is no report created if there are unsolved packages found by resolver.

Comment (pacospace) [reply thoth/storages/graph/postgres.py]: But it depends on the same `document["result"].get("report", {})`
```
            # Mark down packages that were not solved if adviser run failed.
            for unresolved in document["result"].get("report", {}).get("_ERROR_DETAILS", {}).get("unresolved", []):
                self._create_python_package_version(
                    session,
                    package_name=unresolved,
                    package_version=None,
                    index_url=None,
                    os_name=None,
                    os_version=None,
                    python_version=None,
                    sync_only_entity=True,
                )
```

Comment (pacospace) [reply thoth/storages/graph/postgres.py]: But it depends on the same `document["result"].get("report", {})`

```
            # Mark down packages that were not solved if adviser run failed.
            for unresolved in document["result"].get("report", {}).get("_ERROR_DETAILS", {}).get("unresolved", []):
                self._create_python_package_version(
                    session,
                    package_name=unresolved,
                    package_version=None,
                    index_url=None,
                    os_name=None,
                    os_version=None,
                    python_version=None,
                    sync_only_entity=True,
                )
```

**Pull request merged**